pkg/cluster/serviceprincipal.go

```diff
@@ -48,27 +44,21 @@ func (m *manager) clusterSPObjectID(ctx context.Context) error {
 	// wait.PollImmediateUntil. Doing this will not propagate the latest error
 	// to the user in case when wait exceeds the timeout
 	_ = wait.PollImmediateUntil(10*time.Second, func() (bool, error) {
-		var res azgraphrbac.ServicePrincipalObjectResult
-		res, err = m.spApplications.GetServicePrincipalsIDByAppID(ctx, spp.ClientID)
+		clusterSPObjectID, err = utilgraph.GetServicePrincipalIDByAppID(ctx, m.spGraphClient, spp.ClientID)
```

Comment:
For a future refactor, would it make sense to extend the functionality of the spGraphClient to add this function to a subtype?  Then we wouldn't have to pass it, but could make our "own" graphClientAddon and then call this function directly with only passing in a clientId.

Reply:
I would say "maybe" for now (borderline overengineering imo), but "definitely" if there comes a need for more graph utility functions.